### ISSUE TITLE
False negative: `Literal[Ellipsis]`

### ISSUE THREAD

**Issue by smurfix** [python/mypy#18175]
**Bug Report**

Empty classes and similar types (e.g. `Ellipsis`) are often used as placeholders for an uber-None parameter, thus they should be allowed in `Literal[…]`. Currently they are not.

**To Reproduce**

``` python
from typing import Literal,Any
class NotGiven:
    pass
def fn(x:int|None|Literal[NotGiven]=NotGiven) -> Any:
    return 42
print(fn())
```
or

``` python
from typing import Literal,Any
def fn(x:int|None|Literal[Ellipsis]=...) -> Any:
    ...
```

**Expected Behavior**

No error.

**Actual Behavior**

`/tmp/tt.py:4: error: Parameter 1 of Literal[...] is invalid  [valid-type]`

**Your Environment**

- Mypy version used: 1.13.0
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 11

**Comment by erictraut** [comment]
The behavior you're suggesting here is not allowed by the [Python typing spec](https://typing.readthedocs.io/en/latest/spec/index.html) currently. Mypy is correct to flag this as an error.

If you would like to propose an extension or amendment to the typing spec, you are welcome to follow the steps outlined [here](https://github.com/python/typing-council#decisions). There is already a draft PEP relating to [sentinel values](https://peps.python.org/pep-0661/) that is quite far along. It would cover your `NotGiven` use case. It wouldn't cover `Ellipsis`. However, the following is accepted by type checkers (requires Python 3.10 and newer):

```python
from types import EllipsisType

def fn(x: int | None | EllipsisType = ...) -> None: ...
```

**Comment by smurfix** [comment]
Ah. Thank you, I was not aware of these issues.

As the PEP seems to be reasonably far along, is there an effort to teach mypy about it?

**Comment by JelleZijlstra** [comment]
There is no effort that I know of for supporting PEP 661 in mypy. It may be better to wait for a decision on the PEP.
